tests/e2e/utils/customer.js

```diff
@@ -0,0 +1,135 @@
+/**
+ * Helper functions for handling the cart.
+ *
+ * @typedef { import( '@playwright/test' ).Page } Page
+ */
+
+/**
+ * External dependencies
+ */
+const { expect } = require( '@playwright/test' );
+
+/**
+ * Internal dependencies
+ */
+import { LOAD_STATE } from './constants';
+const config = require( '../config/default.json' );
+
+/**
+ * Adds a simple product to the cart.
+ *
+ * @param {Page}   page
+ * @param {number} productID
+ */
+export async function simpleProductAddToCart( page, productID ) {
+	await page.goto( `?p=${ productID }` );
+
+	const addToCart = '.single_add_to_cart_button';
+	await page.locator( addToCart ).first().click();
+	await expect(
+		page.getByText( 'has been added to your cart' )
+	).toBeVisible();
+
+	// Wait till all tracking event request have been sent after page reloaded.
+	await page.waitForLoadState( LOAD_STATE.DOM_CONTENT_LOADED );
+}
+
+/**
+ * Adds a variable product to the cart.
+ *
+ * @param {Page}   page
+ * @param {number} productID
+ */
+export async function variableProductAddToCart( page, productID ) {
+	await page.goto( `?p=${ productID }` );
+
+	// Default attributes are set, so we just need to wait for the add to cart button to be enabled.
+	await page.waitForTimeout( 3000 );
+
+	const addToCart = '.single_add_to_cart_button:not(.disabled)';
+	await page.locator( addToCart ).click();
+
+	await expect(
+		page.getByText( 'has been added to your cart' )
+	).toBeVisible();
+
+	// Wait till all tracking event request have been sent after page reloaded.
+	await page.waitForLoadState( LOAD_STATE.DOM_CONTENT_LOADED );
+}
+
+/**
+ * Adds a related product to the cart.
+ *
+ * @param {Page} page
+ *
+ * @return {number} Product ID of the added product.
+ */
+export async function relatedProductAddToCart( page ) {
+	const addToCart = ( await page.locator( '.related.products' ).isVisible() )
+		? '.related.products .add_to_cart_button'
+		: '.wp-block-woocommerce-related-products .add_to_cart_button';
+
+	const addToCartButton = await page.locator( addToCart ).first();
+	await addToCartButton.click();
+	await expect( addToCartButton.getByText( '1 in cart' ) ).toBeVisible();
+	return await page.$eval( addToCart, ( el ) => el.dataset.product_id );
+}
+
+/**
+ * Add a product to the cart from a block shop page.
+ *
+ * @param {Page}   page
+ * @param {number} productID
+ */
+export async function blockProductAddToCart( page, productID ) {
+	const addToCart = `[data-product_id="${ productID }"]`;
```

Comment:
:nail_care: :nail_care: 
Depending on the product, this selector could match not the "Add to cart" button but the "Select option" button.

From the implementation behavior perspective, I think it's fine. The user of the function probably is aware of what product ID they give, but maybe it could be worth mentioning in a comment.